pkg/infrastructure/azure/azure.go

```diff
@@ -633,7 +633,7 @@ func (p *Provider) PostProvision(ctx context.Context, in clusterapi.PostProvisio
 			bootstrapNicName:     fmt.Sprintf("%s-bootstrap-nic", in.InfraID),
 			frontendIPConfigID:   frontendIPConfigID,
 			inboundNatRuleID:     *inboundNatRule.ID,
-			inboundNatRuleName:   "ssh_in",
+			inboundNatRuleName:   fmt.Sprintf("ssh_in_%s", in.InfraID),
```

Comment:
Nit: Maybe make this `fmt.Sprintf("%s-ssh-in", in.InfraID)` to keep it consistent with naming convention used for resourceGroupName, resourceGroupName and sshRuleName (replace _ with - here too).

Reply:
Agree & updated.